src/components/Form/PollForm.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect } from 'react';
+import { Alert, Button, Form } from 'react-bootstrap';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import * as yup from 'yup';
+
+import { Slide } from 'models/presentation.model';
+import { axiosWithToken } from 'utils';
+import axios from 'axios';
+import { Loader } from 'components/Common';
+
+interface PollFormProps {
+  slideInfo?: Slide;
+}
+
+interface IFormInput {
+  title: string;
+  option1: string;
+  option2: string;
+}
+
+yup.setLocale({
+  mixed: {
+    required: 'This field is required',
+  },
+});
+
+const schema = yup
+  .object({
+    title: yup.string().required(),
+    option1: yup.string().required(),
+    option2: yup.string().required(),
+  })
+  .required();
+
+const PollForm: React.FC<PollFormProps> = ({ slideInfo }) => {
+  const queryClient = useQueryClient();
+  const [errMsg, setErrMsg] = React.useState('');
+  const mutation = useMutation({
+    mutationFn: (data: IFormInput) => {
+      const updateOption1 = axiosWithToken.patch('/slide', {
+        optionId: slideInfo?.pollSlides[1].id,
+        option: data.option1,
+      });
+      const updateOption2 = axiosWithToken.patch('/slide', {
+        optionId: slideInfo?.pollSlides[0].id,
+        option: data.option2,
+      });
+      const updateTitle = axiosWithToken.patch(`/slide/${slideInfo?.id}`, {
+        title: data.title,
+      });
+      return Promise.all([updateOption1, updateOption2, updateTitle]);
+    },
+    onError: (error) => {
+      if (axios.isAxiosError(error)) {
+        setErrMsg(
+          error?.response?.data?.message ||
+            'There was a problem with server. Please try again later.'
+        );
+      }
+    },
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: ['slideDetail', slideInfo?.index],
+      });
+    },
+  });
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setValue,
+  } = useForm<IFormInput>({
+    resolver: yupResolver(schema),
+    defaultValues: {
+      title: slideInfo?.title,
+      option1: slideInfo?.pollSlides[1].option,
+      option2: slideInfo?.pollSlides[0].option,
+    },
+    reValidateMode: 'onChange',
+  });
+
+  const onSubmit: SubmitHandler<IFormInput> = (data) => {
+    mutation.mutate(data);
+  };
+
+  useEffect(() => {
+    setValue('title', slideInfo?.title || '');
+    setValue('option1', slideInfo?.pollSlides[1].option || '');
+    setValue('option2', slideInfo?.pollSlides[0].option || '');
```

Comment:
Why is `pollSlides[0]` set as `option2`?

Reply:
Idk =)))))) để sửa lại =))))